src/index.js

```diff
@@ -0,0 +1,150 @@
+  console.log(allCityDetails[selectedCity].temperature)
+  document.getElementById('tempc-val').innerHTML = allCityDetails[selectedCity].temperature
+  document.getElementById('tempf-val').innerHTML = ((parseInt(allCityDetails[selectedCity].temperature) * 9 / 5) + 32).toFixed(2) + 'F'
+  document.getElementById('humid-val').innerHTML = allCityDetails[selectedCity].humidity
+  document.getElementById('precip-val').innerHTML = allCityDetails[selectedCity].precipitation
+  const time = document.getElementsByClassName('time-text')[0]
+  const date = document.getElementById('date-text')
+  const timeJson = allCityDetails[selectedCity].dateAndTime
+  console.log(timeJson)
+  const separatedArray = timeJson.split(', ')
+  console.log(separatedArray)
+  const months = ['JAN', 'FEB', 'MAR', 'APR', 'MAY', 'JUN', 'JUL', 'AUG', 'SEP', 'OCT', 'NOV', 'DEC']
+  const splitDate = separatedArray[0].split('/')
+  console.log(splitDate)
+  date.innerHTML = splitDate[1] + '-' + months[parseInt(splitDate[0]) - 1] + '-' + splitDate[2]
+  console.log(date)
+  const splitTime = separatedArray[1].split(':')
+  console.log(splitTime)
+  const splitSecAndMeridiem = splitTime[2].split(' ')
+  time.innerHTML = splitTime[0] + ':' + splitTime[1] + ':' + '<small>' + splitSecAndMeridiem[0] + '<small>'
+  const timeState = document.getElementById('time-img')
+  console.log(splitSecAndMeridiem[1])
+  if (splitSecAndMeridiem[1] === 'PM') {
+    timeState.src = '../Assets/HTML&CSS/GeneralImages&Icons/pmState.svg'
+    time.classList.add('pm-state')
+  } else {
+    timeState.src = '../Assets/HTML&CSS/GeneralImages&Icons/amState.svg'
+    time.classList.add('time-text')
+    time.classList.remove('pm-state')
+  }
+  changeTopScroll(allCityDetails[selectedCity], splitTime[0], splitSecAndMeridiem[1])
+}
+
+/**
+ * This function will load the the city names in option of datalist
+ * @param {object} data all city data JSON file
+ */
+function selectOption (data) {
+  console.log(data)
+  data = Object.values(data)
+  const dataList = document.querySelector('.datalist-options')
+  console.log(data)
+  data.map((s) => {
+    const options = document.createElement('option')
+    options.value = s.cityName
+    dataList.appendChild(options)
+    return 0
+  })
+}
+
+/**
+ * This function will fetch data from JSON file using fetch api
+ * @returns {object} cityDetails
+ */
+async function fetchdata () {
+  const cityDetails = await fetch('../Assets/HTML&CSS/files/data.json')
+    .then((res) => { return res.json() })
+    .then((data) => { return data })
+  return cityDetails
+}
+
+/**
+ *  This is  the first function which starts after the page loads and calls the selectoption to load options in datalist
+ */
+async function pageStart () {
+  allCityDetails = await fetchdata()
+  selectOption(allCityDetails)
+  changeValues()
+  document.getElementsByClassName('datalist-options')[0].addEventListener('change', changeValues)
+}
+
+window.addEventListener('load', pageStart)
```

Comment:
Instead of listening at the `load` event, we can use an IIFE wrapped over the whole code

Reply:
If you're needed to wait for the script to load until other dependencies are loaded in HTML, we can also use deferred load method for script tag.